scripts/cbmc_json_parser.py

```diff
@@ -281,15 +285,42 @@ def modify_undefined_function_checks(properties):
     2. If a function with missing definition is reachable, then we turn all SUCCESS status to UNDETERMINED.
     If there are no reachable functions with missing definitions, then the verification is not affected, so we retain all of the SUCCESS status.
     """
-    has_unknown_location_checks = False
+    has_undefined_function_checks = False
     for property in properties:
         # Specifically trying to capture assertions that CBMC generates for functions with missing definitions
         if GlobalMessages.ASSERTION_FALSE in property["description"] and extract_property_class(
                 property) == GlobalMessages.DEFAULT_ASSERTION and not hasattr(property["sourceLocation"], "file"):
             property["description"] = "Function with missing definition is unreachable"
             if property["status"] == "FAILURE":
-                has_unknown_location_checks = True
-    return has_unknown_location_checks
+                has_undefined_function_checks = True
+    return has_undefined_function_checks
+
+def modify_expect_fail_checks(
+        properties,
+        has_failed_sanity_check,
+        has_reachable_undefined_functions,
+        has_failed_unwinding_asserts):
+    """
+    Invert status for Checks with the property_class "expect_fail" from FAILURE TO SUCCESS and vice versa.
+    Modify further depending on unwind/undefined function check statuses.
+    """
+    for property in properties:
```

Comment:
It seems to me that if we call this function after the loop that updates `SUCCESS` to `UNDETERMINED` in `postprocess_results`, the logic can be simplified to "swap SUCCESS with FAILURE".

Reply:
Will do.